servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1094,4 +1120,30 @@ public void reportLostFile(long fileId) throws FileDoesNotExistException {
       return mDependencyMap.getPriorityDependencyList();
     }
   }
+
+  /**
+   * MasterInodeTTL periodic check.
+   */
+  public final class MasterInodeTTLCheckExecutor implements HeartbeatExecutor {
+    @Override
+    public void heartbeat()  {
+      synchronized (mInodeTree) {
+        List<Inode> inodes = mInodeTree.getInodeChildrenRecursive(mInodeTree.getRoot());
```

Comment:
What about maintaining a list of `InodeFile` references with ttl in `MasterInodeTTLCheckExecutor`, each time we create a file with ttl, we add it to the list, and periodically traverse the list to remove those out of time. Then the operation time is O(n) for each periodical check, n is number of files with ttl, but no overhead when creating file.

Or a minimum heap of `InodeFile` references, priority key is the time to be deleted, that is `creationTime + ttl`, then each time an InodeFile with ttl is created, an extra O(logn) is spent, but in each periodic check, we can keep getting minimum element(O(logn)) from the heap until we find a file that should not be deleted, then there is no need to try to delete the remaining files because their time for deletion is even larger. 

If by design, we expect a large fraction of files to be deleted due to ttl in each periodic check, the first solution is preferred, otherwise, the second is better.